packages/agents/src/AgentBase.ts

```diff
@@ -45,7 +51,11 @@ export abstract class AgentBase<TRunArgs, TAgentBaseContext extends AgentBaseCon
         this.context,
         this.config.functions.map((fn) => {
           return {
-            definition: fn,
+            definition: {
```

Comment:
What's this about?

Reply:
Since the functions have been converted to a class with getters (https://github.com/polywrap/evo.ninja/blob/seperate-scripter/packages/agents/src/AgentFunctionBase.ts) we need to manually destructure the object so that its values get copied over.

If we don't do this we hit a runtime error because the function's properties don't exist (because copy didn't succeed).

Reply:
Happy to find a more elegant way of solving this so the UX of the `AgentFunctionBase` class isn't like this.

Reply:
We could add a definition getter to AgentFunctionBase?

Reply:
added